test/e2e/tests/provider-events.spec.js

```diff
@@ -0,0 +1,79 @@
+const { strict: assert } = require('assert');
+const { By, Key, until } = require('selenium-webdriver');
+const { withFixtures, regularDelayMs, largeDelayMs } = require('../helpers');
+
+describe('MetaMask', function () {
+  it('provider listening for events', async function () {
+    const ganacheOptions = {
+      accounts: [
+        {
+          secretKey:
+            '0x7C9529A67102755B7E6102D6D950AC5D5863C98713805CEC576B945B15B71EAC',
+          balance: 25000000000000000000,
+        },
+      ],
+    };
+    await withFixtures(
+      {
+        dapp: true,
+        fixtures: 'connected-state',
+        ganacheOptions,
+        title: this.test.title,
+      },
+      async ({ driver }) => {
+        await driver.navigate();
+        const passwordField = await driver.findElement(By.css('#password'));
+        await passwordField.sendKeys('correct horse battery staple');
+        await passwordField.sendKeys(Key.ENTER);
+        await driver.clickElement(
+          By.css('[data-testid="account-options-menu-button"]'),
+        );
+        await driver.clickElement(
+          By.css('[data-testid="account-options-menu__account-details"]'),
+        );
+        const addressInput = await driver.findElement(
+          By.css('.readonly-input__input'),
+        );
+        const publicAddress = await addressInput.getAttribute('value');
+
+        const accountModal = await driver.findElement(By.css('span .modal'));
+        await driver.clickElement(By.css('.account-modal__close'));
+
+        await driver.wait(until.stalenessOf(accountModal));
+        await driver.delay(regularDelayMs);
```

Comment:
It seems as though the only purpose of this section is to obtain the public address. We don't need to get this interactively upon every test though - we already know it. It's in the fixture: `0x5cfe73b6021e818b776b421b1c4db2474086a7e1`

We can hard-code this instead, to save all of this time.